Build error: UpdateEvent type missing processing_step and gm_response_ready variants

## Bug

`npm run build` fails with TypeScript errors after merging issue #12 changes.

## Error

```
src/stores/chatStore.ts(219,22): error TS2367: This comparison appears to be unintentional because the types '"connected" | "error" | "processing_complete" | "processing_error" | "ping"' and '"processing_step"' have no overlap.
src/stores/chatStore.ts(221,40): error TS2339: Property 'step' does not exist on type 'never'.
src/stores/chatStore.ts(222,22): error TS2367: This comparison appears to be unintentional because the types '"connected" | "error" | "processing_complete" | "processing_error" | "ping"' and '"gm_response_ready"' have no overlap.
src/stores/chatStore.ts(224,30): error TS2339: Property 'message' does not exist on type 'never'.
```

## Root Cause

`chatStore.ts` handles two new SSE event types (`processing_step` and `gm_response_ready`) that were added as part of issue #12, but `UpdateEvent` in `src/services/api.ts` was not updated to include them. TypeScript narrows the type to `never` in the new branches, causing the property accesses (`data.step`, `data.message`) to fail.

## Fix

Add `ProcessingStepEvent` and `GmResponseReadyEvent` interfaces to `src/services/api.ts` and include them in the `UpdateEvent` union type.

--- a/src/services/api.ts
+++ b/src/services/api.ts
@@ -303,7 +303,17 @@ export interface ProcessingErrorEvent {
   timestamp: string;
 }
 
-export type UpdateEvent = ProcessingStartedEvent | ProcessingCompleteEvent | ProcessingErrorEvent | ConnectedEvent | { type: 'ping' } | { type: 'error'; message: string };
+export interface ProcessingStepEvent {
+  type: 'processing_step';
+  step: string;
+}
+
+export interface GmResponseReadyEvent {
+  type: 'gm_response_ready';
+  message: import('../types').Message;
+}
+
+export type UpdateEvent = ProcessingStartedEvent | ProcessingCompleteEvent | ProcessingErrorEvent | ConnectedEvent | ProcessingStepEvent | GmResponseReadyEvent | { type: 'ping' } | { type: 'error'; message: string };
 
 // World updates API (multi-user sync notifications)
 export const updates = {

